opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -37,6 +38,12 @@
 
 namespace Opm {
 
+    /// Returns a copy of s where every letter has been upper cased.
+    static std::string toupper(const std::string& s) {
+        auto ret = std::string(s);
+        std::transform(ret.begin(), ret.end(), ret.begin(), ::toupper);
```

Comment:
This is pretty dangerous. `::toupper` and friends from C can be implemented as macros and sometimes behave surprising when used with std algorithm. The most common solution is to wrap it in a lambda.